src/analyzer.ts

```diff
@@ -417,119 +319,348 @@ export class Analyzer {
           commandAst,
           packageJson.jsonFile
         );
-        if (!result.ok) {
-          return {ok: false, error: [result.error]};
+        if (result.ok) {
+          command = result.value;
+        } else {
+          commandError = true;
+          placeholder.failures.push(result.error);
         }
-        command = result.value;
       }
     }
 
-    let files: undefined | ArrayNode<string>;
-    let output: undefined | ArrayNode<string>;
-    let clean: undefined | JsonAstNode<true | false | 'if-file-deleted'>;
-    if (wireitConfig !== undefined) {
-      if (command === undefined && dependencies.length === 0) {
-        return {
-          ok: false,
-          error: [
-            {
-              type: 'failure',
-              reason: 'invalid-config-syntax',
-              script: placeholder,
-              diagnostic: {
-                severity: 'error',
-                message: `A wireit config must set at least one of "wireit" or "dependencies", otherwise there is nothing for wireit to do.`,
-                location: {
-                  file: packageJson.jsonFile,
-                  range: {
-                    length: wireitConfig.name.length,
-                    offset: wireitConfig.name.offset,
-                  },
-                },
-              },
+    if (
+      wireitConfig !== undefined &&
+      command === undefined &&
+      dependencies.length === 0 &&
+      !dependenciesErrored &&
+      !commandError
+    ) {
+      placeholder.failures.push({
+        type: 'failure',
+        reason: 'invalid-config-syntax',
+        script: placeholder,
+        diagnostic: {
+          severity: 'error',
+          message: `A wireit config must set at least one of "wireit" or "dependencies", otherwise there is nothing for wireit to do.`,
```

Comment:
```suggestion
          message: `A wireit config must set at least one of "command" or "dependencies", otherwise there is nothing for wireit to do.`,
```

Reply:
Done